### PR TITLE
Add Basic Kubernetes Support

### PR DESCRIPTION
**Pull request by spyder-griffith**
This includes support for loading and viewing clusters. It also adds a cache for loading sources and organizations for users and adds more release configurations.

### CODE REVIEW

**Review by brent-spyder — APPROVED** [review]
Nothing glaring stood out to me. I suggest investing some time in building a test framework. That way others can collaborate without fear of breaking something.

File: spydertop/config.py [hunk 1]
```diff
@@ -283,3 +278,14 @@ def is_complete(self):
             and "*" not in self.machine  # * is a wildcard for the source
             and self.start_time is not None
         ) or not isinstance(self.input, str)
+
+
+def get_config_dir() -> Path:
+    """Returns the path to the config directory, ensuring that it exists"""
+    config_dir = os.path.join(os.environ.get("HOME"), ".spyderbat-api/")  # type: ignore
```
Comment (spyder-ian) [inline spydertop/config.py:285]: os.path.join is there to get the slashes the right way,  I think that trailing slash is extra and it wouldn't work on a windows like system if we ever wanted to run it there.  It takes arbitrary argument lists also. os.path.join(home, '.configdir','.configfile')

**Review by spyder-ian — APPROVED** [review]
LGTM

File: spydertop/config.py (hunk 1 above)
Comment (brent-spyder) [reply spydertop/config.py:285]: good catch

**Pull request merged**